AnimatedComponents/src/components/timer/Timer.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { useCallback, useEffect, useMemo } from 'react'
+import { Text, TextInput, TextStyle, TouchableOpacity, View, ViewStyle } from 'react-native'
+import Animated, {
+  Easing,
+  cancelAnimation,
+  useAnimatedProps,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated'
+import Svg, { Circle as SvgCircle } from 'react-native-svg'
+
+import { TimerPreset } from '@constants'
+
+import styles from './timer-styles'
+
+const AnimatedSvgCircle = Animated.createAnimatedComponent(SvgCircle)
+const ReanimatedText = Animated.createAnimatedComponent(TextInput)
+interface ITimerProps {
+  /** buttonStyles: is an optional prop to style the buttons of the timer */
+  buttonStyles?: { container?: ViewStyle; text?: TextStyle }
+  /** circularTimerRadius: is an optional prop that indicates the radius of circular timer */
+  circularTimerRadius?: number
+  /** controls: is an optional prop that indicates whether control buttons are visible or not */
+  controls?: boolean
+  /** duration: is a required prop that indicates the total number of seconds of timer */
+  duration: number
+  /** linearTimerWidth: is an optional prop that indicates the width of the linear timer */
+  linearTimerWidth?: number
+  /** showTimeLeft: is an optional prop that indicates whether to display time left on screen or not. */
+  showTimeLeft?: boolean
+  /** strokeColor: is an optional prop that indicates the color of timer */
+  strokeColor?: string
+  /** timeLeftTextStyle: is an optional prop to style the time left text of the timer */
+  timeLeftTextStyle?: TextStyle
+  /** type: is a required prop that indicates the type of timer, circular or linear */
+  type: TimerPreset
+}
+
+const Timer = (props: ITimerProps) => {
+  const {
+    buttonStyles = {},
+    circularTimerRadius = 45,
+    controls = false,
+    duration,
+    linearTimerWidth = 10,
+    showTimeLeft = true,
+    strokeColor = 'black',
+    timeLeftTextStyle = {},
+    type,
+  } = props
+
+  const progress = useSharedValue(0)
+  const isRunning = useSharedValue(false)
+  const isPaused = useSharedValue(false)
+
+  const handleControl = useCallback(
+    (action: 'start' | 'pause' | 'resume') => {
+      switch (action) {
+        case 'start': {
+          isRunning.value = true
+          isPaused.value = false
+          startAnimation(duration * 1000)
```

Comment:
'startAnimation' was used before it was defined.eslint[@typescript-eslint/no-use-before-define](https://typescript-eslint.io/rules/no-use-before-define)

Reply:
never call a function or variable before its defination